### PR TITLE
Rename Azure driver to Azure Disk driver in e2e test

### PR DESCRIPTION
**Pull request by chewong**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

/kind cleanup

**What this PR does / why we need it**:

Renaming Azure driver to Azure Disk driver to distinguish between Azure Disk and Azure File.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: test/e2e/storage/drivers/in_tree.go [hunk 1]
```diff
@@ -1399,33 +1399,35 @@ func (v *vSphereVolume) DeleteVolume() {
 }
 
 // Azure
```
Comment (kkmsft) [inline test/e2e/storage/drivers/in_tree.go]: nit: Azure ==> Azure disk driver tests

File: test/e2e/storage/drivers/in_tree.go [hunk 2]
```diff
@@ -1441,16 +1443,16 @@ func InitAzureDriver() testsuites.TestDriver {
...
 }
 
-func (a *azureDriver) GetDriverInfo() *testsuites.DriverInfo {
+func (a *azureDiskDriver) GetDriverInfo() *testsuites.DriverInfo {
 	return &a.driverInfo
 }
 
-func (a *azureDriver) SkipUnsupportedTest(pattern testpatterns.TestPattern) {
+func (a *azureDiskDriver) SkipUnsupportedTest(pattern testpatterns.TestPattern) {
 	framework.SkipUnlessProviderIs("azure")
```
Comment (kkmsft) [inline test/e2e/storage/drivers/in_tree.go]: Do we need to skip the AzureFile test as well here ? Or would the future PR on azure file cover this ?

Comment (andyzhangx) [reply test/e2e/storage/drivers/in_tree.go]: azurefile e2e test would be covered by https://github.com/kubernetes-sigs/azurefile-csi-driver/pull/129, @kkmsft, actually the e2e tests in `in_tree.go` do not have lots of coverage.

File: test/e2e/storage/drivers/in_tree.go [hunk 3]
```diff
@@ -1399,33 +1399,35 @@ func (v *vSphereVolume) DeleteVolume() {
...
 
-// InitAzureDriver returns azureDriver that implements TestDriver interface
-func InitAzureDriver() testsuites.TestDriver {
-	return &azureDriver{
+// InitAzureDiskDriver returns azureDiskDriver that implements TestDriver interface
+func InitAzureDiskDriver() testsuites.TestDriver {
+	return &azureDiskDriver{
 		driverInfo: testsuites.DriverInfo{
-			Name:             "azure",
-			InTreePluginName: "kubernetes.io/azure-file",
```
Comment (andyzhangx) [inline test/e2e/storage/drivers/in_tree.go]: haha, original test case is even wrong

**Review by andyzhangx — APPROVED** [review]
/lgtm
/assign @jingxu97 for review, thanks.

File: test/e2e/storage/drivers/in_tree.go [hunk 4]
```diff
@@ -1503,31 +1505,24 @@ func (a *azureDriver) GetDynamicProvisionStorageClass(config *testsuites.PerTest
...
 		Driver:    a,
 		Prefix:    "azure",
 		Framework: f,
 	}, func() {}
 }
 
-func (a *azureDriver) CreateVolume(config *testsuites.PerTestConfig, volType testpatterns.TestVolType) testsuites.TestVolume {
+func (a *azureDiskDriver) CreateVolume(config *testsuites.PerTestConfig, volType testpatterns.TestVolType) testsuites.TestVolume {
 	ginkgo.By("creating a test azure disk volume")
-	if volType == testpatterns.InlineVolume {
```
Comment (jingxu97) [inline test/e2e/storage/drivers/in_tree.go]: this is removed because this testpattern is not supported?

Comment (chewong) [reply test/e2e/storage/drivers/in_tree.go]: This if-statement was causing some pod scheduling error. We are trying to run the e2e test suites in an Azure cluster so we don't have `CloudConfig.Zone` (extracted from `--gce-zone`) populated.

**Comment by saschagrunert** [comment]
Hey @chewong 👋, Im currently looking at the latest [generated releases notes](https://github.com/evillgenius75/sig-release/blob/0cc8b17805dc198561054f0e1498a122320a205f/releases/release-1.18/release-notes-draft.md) and saw that this PR is part of it because of the dot at the end of the `None.`. Do you think you can change the PR description to fix that for us? The next re-generation of the notes will fix that issue on our side later on. :)

**Comment by chewong** [comment]
> Hey @chewong 👋, Im currently looking at the latest [generated releases notes](https://github.com/evillgenius75/sig-release/blob/0cc8b17805dc198561054f0e1498a122320a205f/releases/release-1.18/release-notes-draft.md) and saw that this PR is part of it because of the dot at the end of the `None.`. Do you think you can change the PR description to fix that for us? The next re-generation of the notes will fix that issue on our side later on. :)

Removed the dot.

**Comment by saschagrunert** [comment]
Wonderful, thank you 🙏

**Pull request merged**